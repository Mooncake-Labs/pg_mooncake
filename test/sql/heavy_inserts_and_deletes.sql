-- Create a temporary table for testing
CREATE TABLE test_table (
    id INTEGER,
    data TEXT
) using columnstore;

-- Insert a large number of rows with different strings
INSERT INTO test_table (id, data)
SELECT id, 'Test data ' || id
FROM generate_series(1, 100000) AS id;

-- Simple select
select count(*) from test_table;
select * from test_table limit 10;

-- Update a specific range of rows
UPDATE test_table
SET data = 'Updated data ' || id
WHERE id <= 50 and id >= 60;

select * from test_table where id < 100;

-- Delete a specific number of rows from the table
DELETE FROM test_table
WHERE id <= 999999;

select * from test_table;
-- Drop the temporary table
DROP TABLE test_table;