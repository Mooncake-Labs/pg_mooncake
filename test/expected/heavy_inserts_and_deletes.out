-- Create a temporary table for testing
CREATE TEMPORARY TABLE test_table (
    id SERIAL PRIMARY KEY,
    data TEXT
);
-- Insert a large number of rows with different strings
INSERT INTO test_table (data)
SELECT 'Test data ' || id
FROM generate_series(1, 1000000) AS id;
-- Delete a specific number of rows from the table
DELETE FROM test_table
WHERE id <= 999999;
select * from test_table;
   id    |       data        
---------+-------------------
 1000000 | Test data 1000000
(1 row)

-- Drop the temporary table
DROP TABLE test_table;
